src/components/InputDropdown/index.tsx

```diff
@@ -0,0 +1,353 @@
+'use client';
+
+import {
+  useState,
+  useEffect,
+  useRef,
+  useMemo,
+  KeyboardEvent,
+  Dispatch,
+  SetStateAction,
+} from 'react';
+import {
+  DropdownErrorText,
+  DropdownInput,
+  DropdownInputContainer,
+  DropdownInputLabel,
+  DropdownInputTag,
+  TagContainer,
+} from './styles';
+import DropdownMenu from '../DropdownMenu';
+import { DropdownContainer } from '../FilterDropdown/styles';
+
+interface CommonProps {
+  placeholder?: string;
+  error?: string;
+  disabled?: boolean;
+  label: string;
+  options: Set<string>;
+}
+
+interface MultiSelectProps extends CommonProps {
+  multi: true;
+  value: Set<string>;
+  setValue: Dispatch<SetStateAction<Set<string>>>;
+}
+
+interface SingleSelectProps extends CommonProps {
+  multi?: false;
+  value: string;
+  setValue: Dispatch<SetStateAction<string>>;
+}
+
+type InputDropdownProps = MultiSelectProps | SingleSelectProps;
+
+export default function InputDropdown({
+  placeholder,
+  error = '',
+  disabled = false,
+  label,
+  options,
+  multi,
+  value,
+  setValue,
+}: InputDropdownProps) {
+  const containerRef = useRef<HTMLDivElement>(null);
+  const inputRef = useRef<HTMLParagraphElement>(null);
+  const itemsRef = useRef<Map<string, HTMLParagraphElement | null>>(new Map());
+  const [focusIndex, setFocusIndex] = useState(-1);
+  const [menuVisible, setMenuVisible] = useState(false);
+  const [inputValue, setInputValue] = useState('');
+
+  // store array format for focus indexing
+  const optionsArray = useMemo(
+    () =>
+      Array.from(new Set(options)).filter(o =>
+        o.toLowerCase().includes(inputValue),
+      ),
+    [options, inputValue],
+  );
+
+  // detect clicking outside of menu box
+  useEffect(() => {
+    function globalClickEvent(this: Document, e: MouseEvent) {
+      if (
+        containerRef.current &&
+        containerRef.current.contains(e.target as Node)
+      )
+        return;
+      setMenuVisible(false);
+    }
+
+    document.addEventListener('click', globalClickEvent);
+
+    return () => {
+      document.removeEventListener('click', globalClickEvent);
+    };
+  }, []);
+
+  // handle when a tag gets clicked
+  function handleRemoveOption(option: string) {
```

Comment:
is there some way to make it so that the tag is not deleted when it is clicked, but rather, when the 'X' on the tag is clicked?

Reply:
I think this is more of a question for @davidqing6432 
I personally prefer making it delete when the tag is clicked so users don't have to pinpoint the exact 'X', but I also understand it could be odd that the whole thing disappears when you click on it

also for @davidqing6432 (while you're here) thoughts on adding hover and active effects to the tag? (maybe stretch)

Reply:
agree! can see it going both ways